### PR TITLE
Fix bugs&add some test cases

### PR DESCRIPTION
**Pull request by Shaheen47**

### CODE REVIEW

File: include/Dice/rdf_parser/Parser/Turtle/Grammer.hpp [hunk 1]
```diff
@@ -262,7 +262,8 @@ namespace rdf_parser::Turtle::Grammer {
             sor<DOUBLE, DECIMAL, INTEGER> {
     };
 
-    struct turtleString
+    struct
+    turtleString
```
Comment (bigerl) [inline include/Dice/rdf_parser/Parser/Turtle/Grammer.hpp]: unnecessary change

File: include/Dice/rdf_parser/RDF/Term.hpp [hunk 2]
```diff
@@ -369,4 +370,4 @@ namespace rdf_parser::Turtle {
     }
 };
 
-#endif //RDF_PARSER_TERM_HPP
+#endif
```
Comment (bigerl) [inline include/Dice/rdf_parser/RDF/Term.hpp]: add the coment indicating which if is closed again.

File: include/Dice/rdf_parser/TurtleParser.hpp [hunk 3]
```diff
@@ -60,7 +60,7 @@ namespace rdf_parser::Turtle {
 
             operator bool() { return not done_; }
 
-            const Triple &operator*() { return turtle_parser_.getCurrentTriple(); }
+            Triple operator*() { return turtle_parser_.getCurrentTriple(); }
```
Comment (bigerl) [inline include/Dice/rdf_parser/TurtleParser.hpp]: Why are you not returning a const reference anymore? That may be cheaper. Does getCurrentTriple() return a reference?

**Review by bigerl — CHANGES_REQUESTED** [review]

File: include/Dice/rdf_parser/TurtleParser.hpp (hunk 3 above)
Comment (Shaheen47) [reply include/Dice/rdf_parser/TurtleParser.hpp]: getCurrentTriple() is returning a reference and when the iterator is returning a const reference ,the returned value is a const reference to the current triple on the parser..which means if we have the following code for example 
```c++
TurtleParser<StringParser> parser("@prefix : <http://example.org/elements> .                                                                              \n"
									  "<http://en.wikipedia.org/wiki/Helium>   "
		    						  ":atomicNumber 2 ;"
			  						  " :atomicMass 4.002602 ;"
			                          " :specificGravity 1.663E-4 . ");
	TurtleParser<StringParser>::Iterator iterator = parser.begin();
	ASSERT_TRUE(bool(iterator));
	const Triple &integerNumber = *iterator;
	++iterator;
	const Triple &decimalNumber = *iterator;
	++iterator;
	const Triple &doubleNumber = *iterator;

	// check if they are correct
	ASSERT_EQ(integerNumber.object().getIdentifier(), "\"2\"^^<xsd:integer>");
	ASSERT_EQ(decimalNumber.object().getIdentifier(), "\"4.002602\"^^<xsd:decimal>");
	ASSERT_EQ(doubleNumber.object().getIdentifier(), "\"1.663E-4\"^^<xsd:double>");
```

Then  integerNumber,decimalNumber and doubleNumber will be referring to the latest element(all of them refer to the same element) which located at current_triple in the parser .

Do you suggest a more efficient way to solve this problem ?

Comment (bigerl) [reply include/Dice/rdf_parser/TurtleParser.hpp]: I think the way like shown in your example is fine. 
The user is free to decide. Either they uses the efficient const reference:
```C++
const Triple &integerNumber = *iterator;
```
or the get their local, modifiable copy:
```C++
Triple integerNumber = *iterator;
```

Comment (Shaheen47) [reply include/Dice/rdf_parser/TurtleParser.hpp]: ok.then i will also write this on the documentation

**Review by bigerl — APPROVED** [review]
